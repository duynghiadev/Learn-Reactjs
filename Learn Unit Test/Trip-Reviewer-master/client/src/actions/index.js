import axios from 'axios';

import {
  GET_TRIPS,
  GET_USERS,
  GET_TRIP_W_REVIEWER,
  CLEAR_TRIP_W_REVIEWER,
  GET_TRIP_W_REVIEWER_ERROR,
  USER_REGISTER,
  USER_LOGIN,
  RESET_PASSWORD,
  USER_AUTH,
  ADD_TRIP,
  UPDATE_TRIP,
  UPDATE_USER,
  DELETE_TRIP,
  GET_TRIP,
  GET_TRIP_ERROR,
  CLEAR_NEW_TRIP,
  CLEAR_TRIP,
  GET_USER_REVIEWS,
} from '../constants/action-names';

export function getTrips(limit = 10, start = 0, order = 'asc', list = []) {
  const request = axios.get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_TRIPS,
        payload: {
          list: [...list, ...response.data],
          newTripsCount: response.data.length,
        },
      });
    });
}

export function getTripWithReviewer(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(({ data: tripData }) => {
        const trip = tripData;

        return axios.get(`/api/getReviewer?id=${trip.ownerId}`).then(({ data: reviewerData }) => {
          const response = {
            trip,
            reviewer: reviewerData,
          };

          dispatch({
            type: GET_TRIP_W_REVIEWER,
            payload: response,
          });
        });
      })
      .catch(error =>
        dispatch({
          type: GET_TRIP_W_REVIEWER_ERROR,
          payload: error,
        })
      );
}

export function getUserReviews(userId) {
  const request = axios.get(`/api/getUserReviews?user=${userId}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USER_REVIEWS,
        payload: response.data,
      });
    });
}

export function updateUser(data) {
  const request = axios.post('/api/userUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_USER,
        payload: response.data,
      });
    });
}

export function getTrip(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(response => {
        dispatch({
          type: GET_TRIP,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: GET_TRIP_ERROR,
          payload: error,
        });
      });
}

export function updateTrip(data) {
  const request = axios.post('/api/tripUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_TRIP,
        payload: response.data,
      });
    });
}

export function deleteTrip(id) {
  const request = axios.delete(`/api/tripDelete?id=${id}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: DELETE_TRIP,
        payload: response.data,
      });
    });
}

export function clearTrip() {
  return {
    type: CLEAR_TRIP,
    payload: {
      trip: null,
      updatedTrip: false,
      postDeleted: false,
    },
  };
}

export function clearTripWithReviewer() {
  return {
    type: CLEAR_TRIP_W_REVIEWER,
    payload: {
      trip: null,
      reviewer: null,
    },
  };
}

export function loginUser({ email, password }) {
  const request = axios.post('/api/login', { email, password });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_LOGIN,
        payload: response.data,
      });
    });
}

export function resetPassword({ email }) {
  const request = axios.post('/api/resetPassword', { email });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: RESET_PASSWORD,
        payload: response.data,
      });
    });
}

export function auth() {
  const request = axios.get('/api/auth');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_AUTH,
        payload: response.data,
      });
    });
}

export function addTrip(trip) {
  const request = axios.post('/api/trip', trip);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: ADD_TRIP,
        payload: response.data,
      });
    });
}

export function clearNewTrip() {
  return {
    type: CLEAR_NEW_TRIP,
    payload: null,
  };
}

export function getUsers() {
  const request = axios.get('/api/users');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USERS,
        payload: response.data,
      });
    });
}

export function userRegister(user) {
  const request = axios.post('/api/register', user);

  return dispatch =>
    request.then(({ data }) => {
      dispatch({
        type: USER_REGISTER,
        payload: data,
      });
    });
}
