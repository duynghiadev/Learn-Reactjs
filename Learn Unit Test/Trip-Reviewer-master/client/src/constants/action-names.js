export const GET_TRIPS = 'get-trips';
export const GET_USERS = 'get-users';
export const ADD_TRIP = 'add-trip';
export const UPDATE_TRIP = 'update-trip';
export const UPDATE_USER = 'update-user';
export const DELETE_TRIP = 'delete-trip';
export const GET_TRIP = 'get-trip';
export const GET_TRIP_ERROR = 'get-trip-error';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const GET_TRIP_W_REVIEWER_ERROR = 'get-trip-w-reviewer-error';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const CLEAR_NEW_TRIP = 'clear-new-trip';
export const CLEAR_TRIP = 'clear-trip';
export const USER_REGISTER = 'user-register';
export const USER_LOGIN = 'user-login';
export const RESET_PASSWORD = 'reset-password';
export const USER_AUTH = 'user-auth';
export const GET_USER_REVIEWS = 'get-user-reviews';
