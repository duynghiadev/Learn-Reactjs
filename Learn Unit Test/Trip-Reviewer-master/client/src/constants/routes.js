export const HOME = '/';
export const LOGIN = '/login';
export const RESET = '/reset-password';
export const LOGOUT = '/user/logout';
export const REGISTER = '/user/register';
export const USER_PROFILE = '/user';
export const USER_REVIEWS = '/user/user-reviews';
export const EDIT_PROFILE = '/user/edit-profile';
export const EDIT_POST = '/user/edit-post';
export const ADD_TRIP = '/trip/add';
export const TRIPS = '/trips';
