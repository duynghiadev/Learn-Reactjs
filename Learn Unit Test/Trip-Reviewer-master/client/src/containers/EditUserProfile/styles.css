.edit-user-profile-container {
  margin: auto;
  padding: 50px 0;
}
.edit-user-profile-container .form-content-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
.edit-user-profile-container .left-column,
.edit-user-profile-container .right-column {
  padding: 0 25px;
  width: 40%;
  margin: 0 25px;
  max-width: 400px;
}
.edit-user-profile-container .left-column {
  padding-top: 15px;
}
.edit-user-profile-container .info .title-label {
  display: block;
  font-weight: 500;
  color: #02bcd4;
}
.edit-user-profile-container .danger .title-label {
  display: block;
  color: orange;
}
.edit-user-profile-container .avatar {
  text-align: center;
}
.edit-user-profile-container .info {
  box-sizing: border-box;
  margin: auto;
}
.edit-user-profile-container .info .form-element {
  font-size: 23px;
  font-weight: 300;
  margin-top: 10px;
  color: #a4a4a4;
}
.edit-user-profile-container .danger-zone {
  color: orangered;
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  margin: 10px 0 10px 0;
}
.edit-user-profile-container .disclamer {
  color: darkgray;
  text-align: center;
  font-size: 12px;
  margin-bottom: -5px;
}
.edit-user-profile-container .disclamer-error {
  color: red;
  text-align: center;
  font-size: 14px;
  margin-bottom: -5px;
}
.edit-user-profile-container .disclamer-danger {
  color: darkgray;
  text-align: center;
  font-size: 12px;
  width: 95%;
  margin: 12px auto 30px;
}
.edit-user-profile-container input {
  width: -webkit-fill-available;
  border: solid 1px gray;
  height: 36px;
  margin: 5px 0;
  padding-left: 10px;
  font-size: 16px;
}
.edit-user-profile-container .user-avatar-controls {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  padding: 20px 0;
  margin: auto;
}
.edit-user-profile-container .user-avatar-controls .avatar-button {
  text-align: center;
  cursor: pointer;
  display: block;
  color: white;
  padding: 10px;
  border-radius: 4px;
  width: 30%;
  font-size: 16px;
}
.edit-user-profile-container .user-avatar-controls .update-user-avatar {
  background-color: #02bcd4;
}
.edit-user-profile-container .user-avatar-controls .update-user-avatar:hover {
  transform: scale(1.05);
  background-color: #009eb3;
}
.edit-user-profile-container .user-avatar-controls .update-user-avatar:active {
  transform: scale(1.01);
  background-color: #0495a8;
}
.edit-user-profile-container .user-avatar-controls .delete-user-avatar {
  background-color: orange;
}
.edit-user-profile-container .user-avatar-controls .delete-user-avatar:hover {
  transform: scale(1.05);
  background-color: #e69500;
}
.edit-user-profile-container .user-avatar-controls .delete-user-avatar:active {
  transform: scale(1.01);
  background-color: #cc8500;
}
.edit-user-profile-container .button-link {
  cursor: pointer;
  font-size: 20px;
  padding: 10px 15px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
  margin: 50px auto 0;
  width: 260px;
}
.edit-user-profile-container .button-link:hover {
  transform: scale(1.05);
  background: #009eb3;
}
.edit-user-profile-container .button-link:active {
  transform: scale(1.01);
  background: #0495a8;
}
.edit-user-profile-container .calcel {
  background-color: orange;
  display: inline-block;
  width: 230px;
  margin: 25px auto;
}
.edit-user-profile-container .calcel:hover {
  transform: scale(1.05);
  background: #e69500;
}
.edit-user-profile-container .calcel:active {
  transform: scale(1.01);
  background: #cc8500;
}
.edit-user-profile-container .error {
  color: #ff5722;
  margin-top: 5px;
  text-align: center;
}
.edit-user-profile-container .field-error {
  border: 2px solid #ff5722;
  color: #ff5722;
}
.edit-user-profile-container .avatar {
  width: 144px;
  height: 144px;
  margin: 0 auto 25px auto;
}

@media (max-width: 800px) {
  .edit-user-profile-container .form-content-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .edit-user-profile-container .left-column,
  .edit-user-profile-container .right-column {
    width: 75%;
    margin: auto;
    max-width: 400px;
  }
  .edit-user-profile-container .avatar {
    width: 200px;
    height: 200px;
  }
  .edit-user-profile-container .danger-zone {
    margin: 50px 0 10px 0;
  }
}
