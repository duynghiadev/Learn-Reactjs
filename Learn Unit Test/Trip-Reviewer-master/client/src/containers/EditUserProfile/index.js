import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Progress } from 'react-sweet-progress';
import 'react-sweet-progress/lib/style.css';

import ImageUploader from '../../components/ImageUploader';
import UserAvatar from '../../components/UserAvatar';
import { updateUser } from '../../actions';
import { firebase } from '../../firebase';
import { USER_PROFILE } from '../../constants/routes';

import './styles.css';

class EditUserProfile extends PureComponent {
  state = {
    formData: {
      _id: this.props.users.login.id,
      name: this.props.users.login.name,
      lastname: this.props.users.login.lastname,
      email: this.props.users.login.email,
      avatar: this.props.users.login.avatar,
      oldPassword: '',
      newPassword: '',
      repeatPassword: '',
    },
    hideError: {
      name: false,
      lastname: false,
      email: false,
      avatar: false,
      oldPassword: false,
      newPassword: false,
      repeatPassword: false,
    },
    isUploading: false,
    progress: 0,
    uploadError: null,
  };

  componentWillMount() {
    firebase
      .auth()
      .signInAnonymously()
      .catch(error => {
        console.log(error); // eslint-disable-line no-console
      });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.result && nextProps.result.success) {
      nextProps.history.push(USER_PROFILE);
    } else {
      this.setState({
        hideError: {
          name: false,
          lastname: false,
          email: false,
          avatar: false,
          oldPassword: false,
          newPassword: false,
          repeatPassword: false,
        },
      });
    }
  }

  onUploadStarts = () => {
    this.setState({
      isUploading: true,
      progress: 0,
      uploadError: null,
    });
  };

  onUploadProgress = progress => {
    this.setState({ progress });
  };

  onUploadSuccess = filename => {
    const newFormData = { ...this.state.formData };

    newFormData.avatar = filename;

    this.setState({
      formData: newFormData,
      isUploading: false,
      progress: 100,
    });
  };

  onUploadError = error => {
    this.setState({ isUploading: false, uploadError: error });
  };

  getErrorClass(fieldName) {
    return this.formFieldHasError(fieldName) ? 'field-error' : '';
  }

  handleDeleteAvatar = () => {
    const newFormData = { ...this.state.formData };

    newFormData.avatar = null;

    this.setState({ formData: newFormData });
  };

  handleInput = event => {
    const newFormData = { ...this.state.formData };
    const newHideError = { ...this.state.hideError };
    const { value, name } = event.target;

    newFormData[name] = value;
    newHideError[name] = true;

    this.setState({
      formData: newFormData,
      hideError: newHideError,
    });
  };

  submitForm = event => {
    event.preventDefault();
    this.props.dispatch(updateUser(this.state.formData));
  };

  formFieldHasError(fieldName) {
    const { result } = this.props;
    const { hideError } = this.state;
    const errors = result && result.error && result.error.errors;

    return errors && errors[fieldName] && !hideError[fieldName];
  }

  renderError(fieldName) {
    if (this.formFieldHasError(fieldName)) {
      const error = this.props.result.error.errors[fieldName];

      return <div className="error">{error.message}</div>;
    }
    return null;
  }

  renderAvatarControls() {
    const { isUploading, progress } = this.state;

    if (isUploading) {
      return (
        <div className="user-avatar-controls">
          <Progress percent={progress} />
        </div>
      );
    }
    return (
      <div className="user-avatar-controls">
        <ImageUploader
          filename={this.props.users.login.id}
          onUploadStarts={this.onUploadStarts}
          onUploadProgress={this.onUploadProgress}
          onUploadSuccess={this.onUploadSuccess}
          onUploadError={this.onUploadError}
          className="avatar-button update-user-avatar"
        >
          Update
        </ImageUploader>
        <label className="avatar-button delete-user-avatar" onClick={this.handleDeleteAvatar}>
          Delete
        </label>
      </div>
    );
  }

  renderSoftInfoInputs() {
    const { name, lastname } = this.state.formData;

    return (
      <div className="info">
        <div className="form-element">
          <span className="title-label">Name:</span>
          <input
            type="text"
            className={this.getErrorClass('name')}
            name="name"
            placeholder="Enter name"
            value={name}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('name')}

        <div className="form-element">
          <span className="title-label">Lastname:</span>
          <input
            type="text"
            className={this.getErrorClass('lastname')}
            name="lastname"
            placeholder="Enter lastname"
            value={lastname}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('lastname')}
      </div>
    );
  }

  renderDangerZoneInputs() {
    const { oldPassword, newPassword, repeatPassword, email } = this.state.formData;

    return (
      <div className="info danger">
        <div className="form-element">
          <span className="title-label">Current Password:</span>
          <input
            type="password"
            className={this.getErrorClass('oldPassword')}
            name="oldPassword"
            placeholder="Enter your current password"
            value={oldPassword}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('oldPassword')}

        <div className="form-element">
          <span className="title-label">New Password:</span>
          <input
            type="password"
            className={this.getErrorClass('newPassword')}
            name="newPassword"
            placeholder="Enter your new password"
            value={newPassword}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('newPassword')}

        <div className="form-element">
          <span className="title-label">Repeat New Password:</span>
          <input
            type="password"
            className={this.getErrorClass('repeatPassword')}
            name="repeatPassword"
            placeholder="Repeat your new password"
            value={repeatPassword}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('repeatPassword')}

        <div className="form-element">
          <span className="title-label">Email:</span>
          <input
            type="email"
            className={this.getErrorClass('email')}
            name="email"
            placeholder="Enter new email"
            value={email}
            onChange={this.handleInput}
          />
        </div>
        {this.renderError('email')}
      </div>
    );
  }

  renderDisclaimer() {
    if (this.state.uploadError) {
      return (
        <p className="disclamer-error">
          Upload was not successful.
          <br />
          The avatar image should be smaller than <b>1MB</b>.
        </p>
      );
    }
    return (
      <p className="disclamer">
        * To be sure that your avatar will be updated correctly <br />
        press <b>&quot;Submit changes&quot;</b> button at the bottom of the page
      </p>
    );
  }

  renderFormContent() {
    const { avatar } = this.state.formData;

    return (
      <div className="form-content-wrapper">
        <div className="left-column">
          <div className="avatar">
            <UserAvatar filename={avatar} />
          </div>

          {this.renderDisclaimer()}
          {this.renderAvatarControls()}
          {this.renderSoftInfoInputs()}
        </div>

        <div className="right-column">
          <p className="danger-zone">Danger zone</p>
          <p className="disclamer-danger">
            * To change anything from the &quot;Danger Zone&quot; you should enter your&nbsp;
            <b>current password</b>. If you do not enter your current password - nothing will be
            changed.
          </p>

          {this.renderDangerZoneInputs()}
          {this.renderError()}
        </div>
      </div>
    );
  }

  render() {
    return (
      <div className="edit-user-profile-container">
        <form onSubmit={this.submitForm}>
          {this.renderFormContent()}

          <div className="text-center">
            <button type="submit" className="button-link">
              Submit changes
            </button>
          </div>

          <div className="text-center">
            <Link to={USER_PROFILE} className="button-link calcel">
              Cancel
            </Link>
          </div>
        </form>
      </div>
    );
  }
}

EditUserProfile.propTypes = {
  users: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  result: PropTypes.object,
};

EditUserProfile.defaultProps = {
  result: undefined,
};

function mapStateToProps(state) {
  return {
    result: state.users.userUpdate,
  };
}

export default connect(mapStateToProps)(EditUserProfile);
