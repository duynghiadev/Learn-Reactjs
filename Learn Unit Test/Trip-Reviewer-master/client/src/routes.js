import React from 'react';
import { Switch, Route } from 'react-router-dom';

import NotFound from './components/404';
import Home from './containers/Home';
import User from './components/User';
import EditUserProfile from './containers/EditUserProfile';
import Logout from './components/Logout';
import Login from './containers/Login';
import ResetPass from './containers/ResetPass';
import Register from './containers/Register';
import AddTrip from './containers/AddTrip';
import EditTrip from './containers/EditTrip';
import TripView from './containers/TripView';
import UserReviews from './containers/UserReviews';

import Layout from './hoc/layout';
import Auth from './hoc/auth';

import * as routes from './constants/routes';

const Routes = () => (
  <Layout>
    <Switch>
      <Route path={routes.HOME} exact component={Auth(Home, null)} />
      <Route path={routes.LOGIN} exact component={Auth(Login, false)} />
      <Route path={routes.RESET} exact component={Auth(ResetPass, false)} />
      <Route path={routes.LOGOUT} exact component={Auth(Logout, true)} />
      <Route path={routes.REGISTER} exact component={Auth(Register, false)} />
      <Route path={routes.USER_PROFILE} exact component={Auth(User, true)} />
      <Route path={routes.USER_REVIEWS} exact component={Auth(UserReviews, true)} />
      <Route path={`${routes.EDIT_PROFILE}`} exact component={Auth(EditUserProfile, true)} />
      <Route path={`${routes.EDIT_POST}/:id`} exact component={Auth(EditTrip, true)} />
      <Route path={routes.ADD_TRIP} exact component={Auth(AddTrip, true)} />
      <Route path={`${routes.TRIPS}/:id`} exact component={Auth(TripView, null)} />
      <Route path="*" component={Auth(NotFound, null)} />
    </Switch>
  </Layout>
);

export default Routes;
