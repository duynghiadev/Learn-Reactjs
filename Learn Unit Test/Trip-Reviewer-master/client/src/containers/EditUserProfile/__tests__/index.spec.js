import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import EditUserProfile from '..';
import { UPDATE_USER } from '../../../constants/action-names';
import { updateUser } from '../../../actions';
import { USER_PROFILE } from '../../../constants/routes';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('./../../../components/ImageUploader', () => 'ImageUploader');
jest.mock('./../../../components/UserAvatar', () => 'UserAvatar');
jest.mock('./../../../actions', () => ({
  updateUser: jest.fn(() => ({
    type: UPDATE_USER,
    payload: {},
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <EditUserProfile {...props} store={store} />;
};

describe('<EditUserProfile />', () => {
  it('should render component', () => {
    const initialState = {
      users: {},
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with error', () => {
    const initialState = {
      users: {
        userUpdate: {
          success: false,
          error: {
            errors: {
              name: { message: 'test' },
              lastname: { message: 'test' },
              email: { message: 'test' },
              oldPassword: { message: 'test' },
              newPassword: { message: 'test' },
              repeatPassword: { message: 'test' },
            },
          },
        },
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const tree = create(mockComponent(initialState, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should redirect when componentWillReceiveProps is called with success true', () => {
    const push = jest.fn();
    const path = USER_PROFILE;
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const nextProps = {
      result: {
        success: true,
      },
      history: {
        push,
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.componentWillReceiveProps(nextProps);

    expect(push).toHaveBeenCalledWith(path);
  });

  it('should not redirect when componentWillReceiveProps is called with success false', () => {
    const push = jest.fn();
    const path = USER_PROFILE;
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const nextProps = {
      result: {
        success: false,
        message: 'test',
      },
      history: {
        push,
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.componentWillReceiveProps(nextProps);

    expect(push).not.toHaveBeenCalledWith(path);
  });

  it('should dispatch updateUser when submitForm is called', () => {
    const preventDefault = jest.fn();
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const state = {
      name: 'name',
      lastname: 'lastname',
      email: 'email',
    };
    const event = { preventDefault };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();
    instance.setState({ formData: state });

    instance.submitForm(event);

    expect(preventDefault).toHaveBeenCalled();
    expect(updateUser).toHaveBeenCalledWith(state);
  });

  it('handleInput should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
        },
      },
    };
    const event = {
      target: {
        value: 'test',
        name: 'title',
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.handleInput(event);

    expect(instance.state.formData.title).toEqual('test');
  });

  it('handleDeleteAvatar should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
          avatar: 'avatar',
        },
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.handleDeleteAvatar();

    expect(instance.state.formData.avatar).toEqual(null);
  });

  it('onUploadSuccess should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
          avatar: 'avatar',
        },
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.onUploadSuccess('test');

    expect(instance.state.formData.avatar).toEqual('test');
    expect(instance.state.isUploading).toEqual(false);
    expect(instance.state.progress).toEqual(100);
  });

  it('onUploadStarts should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
          avatar: 'avatar',
        },
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.onUploadStarts();

    expect(instance.state.isUploading).toEqual(true);
    expect(instance.state.progress).toEqual(0);
  });

  it('onUploadProgress should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
          avatar: 'avatar',
        },
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.onUploadProgress('progress');

    expect(instance.state.progress).toEqual('progress');
  });

  it('onUploadError should change state correctly', () => {
    const initialState = {
      users: {
        userUpdate: {},
      },
    };
    const props = {
      users: {
        login: {
          id: 'id',
          avatar: 'avatar',
        },
      },
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.onUploadError('error');

    expect(instance.state.uploadError).toEqual('error');
    expect(instance.state.isUploading).toEqual(false);
  });
});
