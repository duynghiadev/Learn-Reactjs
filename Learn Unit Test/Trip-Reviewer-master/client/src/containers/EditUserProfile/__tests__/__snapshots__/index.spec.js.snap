// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditUserProfile /> should render component 1`] = `
<div
  className="edit-user-profile-container"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-content-wrapper"
    >
      <div
        className="left-column"
      >
        <div
          className="avatar"
        >
          <UserAvatar
            filename={undefined}
          />
        </div>
        <p
          className="disclamer"
        >
          * To be sure that your avatar will be updated correctly 
          <br />
          press 
          <b>
            "Submit changes"
          </b>
           button at the bottom of the page
        </p>
        <div
          className="user-avatar-controls"
        >
          <ImageUploader
            className="avatar-button update-user-avatar"
            filename="id"
            onUploadError={[Function]}
            onUploadProgress={[Function]}
            onUploadStarts={[Function]}
            onUploadSuccess={[Function]}
          >
            Update
          </ImageUploader>
          <label
            className="avatar-button delete-user-avatar"
            onClick={[Function]}
          >
            Delete
          </label>
        </div>
        <div
          className="info"
        >
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Name:
            </span>
            <input
              className=""
              name="name"
              onChange={[Function]}
              placeholder="Enter name"
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Lastname:
            </span>
            <input
              className=""
              name="lastname"
              onChange={[Function]}
              placeholder="Enter lastname"
              type="text"
              value={undefined}
            />
          </div>
        </div>
      </div>
      <div
        className="right-column"
      >
        <p
          className="danger-zone"
        >
          Danger zone
        </p>
        <p
          className="disclamer-danger"
        >
          * To change anything from the "Danger Zone" you should enter your 
          <b>
            current password
          </b>
          . If you do not enter your current password - nothing will be changed.
        </p>
        <div
          className="info danger"
        >
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Current Password:
            </span>
            <input
              className=""
              name="oldPassword"
              onChange={[Function]}
              placeholder="Enter your current password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              New Password:
            </span>
            <input
              className=""
              name="newPassword"
              onChange={[Function]}
              placeholder="Enter your new password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Repeat New Password:
            </span>
            <input
              className=""
              name="repeatPassword"
              onChange={[Function]}
              placeholder="Repeat your new password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Email:
            </span>
            <input
              className=""
              name="email"
              onChange={[Function]}
              placeholder="Enter new email"
              type="email"
              value={undefined}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        className="button-link"
        type="submit"
      >
        Submit changes
      </button>
    </div>
    <div
      className="text-center"
    >
      <Link
        className="button-link calcel"
        to="/user"
      >
        Cancel
      </Link>
    </div>
  </form>
</div>
`;

exports[`<EditUserProfile /> should render component with error 1`] = `
<div
  className="edit-user-profile-container"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-content-wrapper"
    >
      <div
        className="left-column"
      >
        <div
          className="avatar"
        >
          <UserAvatar
            filename={undefined}
          />
        </div>
        <p
          className="disclamer"
        >
          * To be sure that your avatar will be updated correctly 
          <br />
          press 
          <b>
            "Submit changes"
          </b>
           button at the bottom of the page
        </p>
        <div
          className="user-avatar-controls"
        >
          <ImageUploader
            className="avatar-button update-user-avatar"
            filename="id"
            onUploadError={[Function]}
            onUploadProgress={[Function]}
            onUploadStarts={[Function]}
            onUploadSuccess={[Function]}
          >
            Update
          </ImageUploader>
          <label
            className="avatar-button delete-user-avatar"
            onClick={[Function]}
          >
            Delete
          </label>
        </div>
        <div
          className="info"
        >
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Name:
            </span>
            <input
              className="field-error"
              name="name"
              onChange={[Function]}
              placeholder="Enter name"
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Lastname:
            </span>
            <input
              className="field-error"
              name="lastname"
              onChange={[Function]}
              placeholder="Enter lastname"
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
        </div>
      </div>
      <div
        className="right-column"
      >
        <p
          className="danger-zone"
        >
          Danger zone
        </p>
        <p
          className="disclamer-danger"
        >
          * To change anything from the "Danger Zone" you should enter your 
          <b>
            current password
          </b>
          . If you do not enter your current password - nothing will be changed.
        </p>
        <div
          className="info danger"
        >
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Current Password:
            </span>
            <input
              className="field-error"
              name="oldPassword"
              onChange={[Function]}
              placeholder="Enter your current password"
              type="password"
              value=""
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              New Password:
            </span>
            <input
              className="field-error"
              name="newPassword"
              onChange={[Function]}
              placeholder="Enter your new password"
              type="password"
              value=""
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Repeat New Password:
            </span>
            <input
              className="field-error"
              name="repeatPassword"
              onChange={[Function]}
              placeholder="Repeat your new password"
              type="password"
              value=""
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
          <div
            className="form-element"
          >
            <span
              className="title-label"
            >
              Email:
            </span>
            <input
              className="field-error"
              name="email"
              onChange={[Function]}
              placeholder="Enter new email"
              type="email"
              value={undefined}
            />
          </div>
          <div
            className="error"
          >
            test
          </div>
        </div>
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        className="button-link"
        type="submit"
      >
        Submit changes
      </button>
    </div>
    <div
      className="text-center"
    >
      <Link
        className="button-link calcel"
        to="/user"
      >
        Cancel
      </Link>
    </div>
  </form>
</div>
`;
